server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -51,6 +51,8 @@ public final class IngestDocument {
 
     private static final String PIPELINE_CYCLE_ERROR_MESSAGE = "Cycle detected for pipeline: ";
     static final String TIMESTAMP = "timestamp";
+    // This is the maximum number of nested pipelines that can be within a pipeline. If there are more, we bail out with an error
+    private static final int MAX_PIPELINES = Integer.parseInt(System.getProperty("ingest.max_pipelines", "100"));
```

Comment:
A very minor nit, but I think `es.ingest.max_pipelines` would be just a little better.